### DISCUSSION TITLE
Mess with secondary layout

### DISCUSSION THREAD

**Discussion by LightBeamOvHe** [cafali/SnapKey#98] [New Features & Ideas]
Hi,

I use an azerty-NF layout and have a US layout installed that I swap depending on when I need it

I noticed that when I use azerty by default and then swap to qwerty, A and Q and mirrored and always press A, while Z and W always press W. Azerty works fine.

When I start Windows with US as default, it's now the azerty layout that's getting weird.

Anyway, I noticed that right at boot it was working perfectly until some programs booted and ... SnapKey was the culprit. Disabling it makes it work perfectly fine.

I loved how I could just lauch it and forget, but now I realize I won't be able to because of that issue :(
I guess the best fix would be to have it not work at all time, so either with only a specific layout or through a list of games

**Comment by minteeaa** [comment]
i can imagine this happens due to the internal keymap being changed, but snapkey not pulling that new update or refreshing the keymap (though i'm not sure if this *should* be automatic or not by the windows hook)

if this only happens when hot-switching layouts, it's more than likely that; i'll see if i can verify if there isn't anyone else to

does restarting snapkey after swapping - [in the latest release](https://github.com/cafali/SnapKey/releases/tag/v1.2.5) - mitigate the issue?

**Comment by LightBeamOvHe** [comment]
Hi, sorry for answering that late

Just tried and yes, it seems that restarting SnapKey after swapping does work properly. I get the azerty keys to repeat even in a US layout, just like it should because I haven't remapped the keys in the config file.

For now I just settled to just launching SnapKey when I play Overwatch, it's more convenient for me. It's a very minor issue and because I don't have any dev skills I let y'all judge if it's worth fixing or not.

**Comment by minteeaa** [comment]
it would make sense for SnapKey to detect keymap switches and re-load the proper mappings for it; I personally use multiple keyboard layouts that would benefit from that change

potentially a good addition, we'll see if it gets around to being done (:
